north/src/api.rs

```diff
@@ -11,14 +26,35 @@ pub struct Message {
     pub payload: Payload,
 }
 
-#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+impl Message {
+    pub fn new(payload: Payload) -> Message {
+        Message {
+            id: uuid::Uuid::new_v4().to_string(),
+            payload,
+        }
+    }
+
+    pub fn new_request(request: Request) -> Message {
+        Message::new(Payload::Request(request))
+    }
+
+    pub fn new_response(respone: Response) -> Message {
+        Message::new(Payload::Response(respone))
+    }
+
+    pub fn new_notification(notification: Notification) -> Message {
+        Message::new(Payload::Notification(notification))
+    }
+}
+
+#[derive(new, Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
 pub enum Payload {
     Request(Request),
     Response(Response),
     Notification(Notification),
 }
 
-#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+#[derive(new, Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
```

Comment:
Is that a preparation for backward compatibility?

Reply:
the `#[derive(new)]`

Reply:
No. It's just to generate `new` methods be cause I'm lazy ;-)